File: 2/2_1.py
# Multiply your final horizontal position by your final depth

horizontal_pos = 0
depth = 0

with open('2.txt', 'r') as f:
    for line in f:
        reading = line.split(' ')
        movement_type = reading[0]
        movement = int(reading[1])

        # depth
        if movement_type == 'down':
            depth += movement
        elif movement_type == 'up':
            depth -= movement

        # horizontal
        else:
            horizontal_pos += movement

print(
    f'With {horizontal_pos=} and {depth=}, final calculation is {horizontal_pos * depth}'
)

File: day_2/2_1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	horizontal_pos := 0
	depth := 0

	f, _ := os.Open("2.txt")
	scanner := bufio.NewScanner(f)

	for scanner.Scan() {
		line := scanner.Text()
		reading := strings.Split(line, " ")
		movement_type := reading[0]
		movement_str := reading[1]
		movement, _ := strconv.Atoi(movement_str)

		// depth
		if movement_type == "down" {
			depth += movement
		} else if movement_type == "up" {
			depth -= movement
		} else {
			horizontal_pos += movement
		}
	}

	calculation := horizontal_pos * depth
	fmt.Println(calculation)
}
